### PR TITLE
BLD: Create OpenBLAS shared object

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @HaoZeke,

This PR builds shared library for OpenBLAS, alongside static library.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -537,14 +540,22 @@ subdir('driver/others')
...
+                            _l2_driver,
+                            _l3_driver,
+                            _others,
+                            _kern,
+                           ],
+                           install: true,
+                          )
+
+_openblas_so = shared_library('openblas',
+                              link_whole: [_openblas],
```
Comment (HaoZeke) [inline meson.build]: Perhaps using `both_libs` is more idiomatic?
https://mesonbuild.com/Reference-manual_functions.html#both_libraries

File: lapack-netlib/SRC/meson.build [hunk 2]
```diff
@@ -501,3 +501,11 @@ netlib_lapack = library(_netlib_lapack_name,
                version: lapack_version,
                soversion: lapack_major_version,
                install: true)
+
+pkg.generate(netlib_lapack,
+             name: 'lapack',
+             filebase: 'meson-lapack',
+             description: 'lapack via meson build',
+             version: f'@pkg_ver@_lapack',
+             install_dir: '../../pkgconfig',
```
Comment (HaoZeke) [inline lapack-netlib/SRC/meson.build]: Can this be replaced with a project scoped variable?

**Review by HaoZeke — CHANGES_REQUESTED** [review]

File: lapack-netlib/SRC/meson.build (hunk 2 above)
Comment (mtsokol) [reply lapack-netlib/SRC/meson.build]: Done!

File: meson.build (hunk 1 above)
Comment (mtsokol) [reply meson.build]: Sure! I used `both_libraries` function instead.

**Review by HaoZeke — APPROVED** [review]

**Comment by HaoZeke** [comment]
Thanks @mtsokol

**Pull request merged**